Renee updates

olson59 opened the pull request:
Changes:

- Pyranda is now dumping mixedness, "mixedness is a common metric for mix modeling that is based on the variance"
- I created two versions of the LC yaml. One for generating simulation data to train the GP model, and one for generating fake experimental data at a higher resolution and including random noise perturbations in the initial conditions. This can be generalized to the AWS yaml.
- There is a csv file for Xiao of the simulation data to train the GP's in Psuade. 200+ samples by 4 columns. The first 3 columns are inputs ['atwood', 'velocity', 'time'], and the last column is the output 'mix_width'. 
- There are two pgen files. simulation_pgen.py creates samples with a Latinhypercube design, and experimental_pgen.py creates samples at a higher resolution with noise. 
- plot_study.py is useful to create plots to view the data and fit a GP.

doutriaux1 committed:
small edits to notebook for interaction when demoing

jmoreno45 committed:
Moreno45/feature/kosh task

jmoreno45 committed:
Merge branch 'moreno45/feature/kosh_task' into 'main'

Moreno45/feature/kosh task

See merge request weave/weave_demos!13

doutriaux1 committed:
Merge branch 'teaser' into 'main'

small edits to notebook for interaction when demoing

See merge request weave/weave_demos!14

jmoreno45 committed:
Merge branch 'moreno45/feature/encore_optimization' into 'main'

Moreno45/feature/encore optimization

See merge request weave/weave_demos!15

jmoreno45 committed:
Moreno45/feature/encore optimization

jmoreno45 committed:
Merge branch 'moreno45/feature/encore_optimization' into 'main'

Moreno45/feature/encore optimization

See merge request weave/weave_demos!16

jmoreno45 committed:
Moreno45/feature/encore optimization

jmoreno45 committed:
Merge branch 'moreno45/feature/encore_optimization' into 'main'

Moreno45/feature/encore optimization

See merge request weave/weave_demos!17

jmoreno45 committed:
Moreno45/feature/encore optimization

jmoreno45 committed:
Moreno45/feature/threadsafe

jmoreno45 committed:
Merge branch 'moreno45/feature/threadsafe' into 'main'

Moreno45/feature/threadsafe

See merge request weave/weave_demos!18

jmoreno45 committed:
Moreno45/feature/threadsafe

jmoreno45 committed:
Merge branch 'moreno45/feature/threadsafe' into 'main'

Moreno45/feature/threadsafe

See merge request weave/weave_demos!20

jmoreno45 committed:
Moreno45/feature/clean up tutorial

jmoreno45 committed:
Merge branch 'moreno45/feature/clean_up_tutorial' into 'main'

Moreno45/feature/clean up tutorial

See merge request weave/weave_demos!21

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/README.md:
```diff
@@ -0,0 +1,22 @@
+# Rayleigh Taylor Modeling with pyranda
+
+pyranda is an mpi parallel high order finited difference solver for arbitrary hyperbolic PDE systems (INSERT LINK).  This set of examples works through simulations of the Rayleigh-Taylor problem,
```
Do we know which link we need to insert here yet?

olson59 replied:
https://github.com/LLNL/pyranda
Are you looking for something like this?

bgunnar5 replied:
Yeah I think linking to the pyranda project itself would work.

jwhite242 replied:
also, no need to fix that in this one: that's a mess i put in there, lol.  the whole readme needs a bit more filling out on the nominal behaviors like some snapshots/curves and stuff too to give a bit of a view of it without running everything

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/bayesian_inference.py:
```diff
@@ -0,0 +1,48 @@
+with h5py.File(fileName, "w") as f:
+    f.create_dataset("inputs", data=X)
+    f.create_dataset("outputs", data=Y)
+    f.close()
```
I don't think `f.close()` is necessary if we're using the `with h5py.File(..)` context manager (see [Closing Files](https://docs.h5py.org/en/stable/high/file.html#closing-files))

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/bayesian_inference.py:
```diff
@@ -0,0 +1,48 @@
+                                 inputs_low=[0.0]*3,
+                                 inputs_high=[1.0]*3,
+                                 proposal_sigmas=[0.2]*3,
+                                 prior=,
```
I'm not familiar with this setting, is it purposefully left blank?

olson59 replied:
The default is a uniform distribution and I need to figure out what to put there to trigger the default, or I can just provide uniform myself. This bayesian inference file is not done yet.

bgunnar5 replied:
Ah ok, sorry for jumping the gun on this then.

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/experimental_pgen.py:
```diff
@@ -0,0 +1,36 @@
+
+# Generate sample points for the parameters from normal distributions
+# for our fake experimental data
+Nexperiments = 8
```
Should we make this a variable in the Maestro spec that we can vary? Variables from the `env` block can be read into the pgen file (instructions for how this is done can be found [here](https://maestrowf.readthedocs.io/en/latest/Maestro/parameter_specification.html#referencing-values-from-a-specifications-env-block)).

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/plot_study.py:
```diff
@@ -0,0 +1,88 @@
+
+# This is single mode cases (too easy for GP's)
+# dataDir = "RT_STUDIES/rayleigh_taylor_pyranda_20240813-113206/run-pyranda/*"
+dataDir = "RT_STUDIES/rayleigh_taylor_pyranda_20240813-125700/run-pyranda/*"
```
Should this be an argument that the user can pass into this script that way it's not hardcoded in?

bgunnar5 commented on CZ/pyranda_rayleigh_taylor/rayleigh_taylor.py:
```diff
@@ -0,0 +1,646 @@
+    headless = args.headless
+
+    ## Define a mesh
+    if is2D:
+        widthToHeight = 4.0
+        xwidth = widthToHeight*numpy.pi
+        problem = "RAYLEIGH_TAYLOR_2D"
+        imesh = (
+            """
+        xdom = (-xwidth/2.0, xwidth/2.0 , int(Npts*widthToHeight), periodic=False)
+        ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        zdom = (0.0, 2*pi*FF,  1, periodic=True)
+        """.replace(
+                "Npts", str(Npts)
+            )
+            .replace("pi", str(numpy.pi))
+            .replace("FF", str(float(Npts - 1) / Npts))
+            .replace("xwidth", str(xwidth))
+            .replace("widthToHeight",str(widthToHeight))
+        )
+        waveLength = 4
+    else:
+        widthToHeight = 3.0
+        xwidth = widthToHeight*numpy.pi
+        problem = "RAYLEIGH_TAYLOR_3D"
+        imesh = (
+            """
+        xdom = (-xwidth/2.0, xwidth/2.0, int(Npts*widthToHeigth), periodic=False)
+        ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        zdom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        """.replace(
+                "Npts", str(Npts)
+            )
+            .replace("pi", str(numpy.pi))
+            .replace("FF", str(float(Npts - 1) / Npts))
+            .replace("xwidth", str(xwidth))
+        )
+        waveLength = 4
```
Wondering if we should just change this to a single string for imesh that's located outside of the if/else since they're so similar. Something like:
```python
if is2D:
    widthToHeight = 4.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_2D"
    waveLength = 4
    zpts = 1
else:
    widthToHeight = 3.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_3D"
    waveLength = 4
    zpts = Npts

imesh = (
    """
    xdom = (-xwidth/2.0, xwidth/2.0, int(Npts*widthToHeight), periodic=False)
    ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
    zdom = (0.0, 2*pi*FF,  zpts, periodic=True)
    """.replace(
            "Npts", str(Npts)
        )
        .replace("pi", str(numpy.pi))
        .replace("FF", str(float(Npts - 1) / Npts))
        .replace("xwidth", str(xwidth))
        .replace("widthToHeight",str(widthToHeight))
        .replace("zpts",str(zpts))
    )
```

doutriaux1 reviewed (approved):
Awesome. Thx Renee

bgunnar5 reviewed (commented):
Looks great, thanks for the work on this Renee! I just had a few small comments on some things here, nothing major.

jwhite242 commented on CZ/pyranda_rayleigh_taylor/rayleigh_taylor_lc_build_venv_sim.yaml:
```diff
@@ -0,0 +1,120 @@
+
+
+study:
+    # - name: make-env-lc
```
want to uncomment these make-env steps given they're the 'build_venv' variants of the studies?

jwhite242 reviewed (approved):
All looks pretty good.  Just a few over arching comments, which don't necessarily need fixing right now:

* for the eventual uq variants we probably want to change the study names and descriptions as well, and maybe even output_path variables?
* for the longer term (and this really doesn't matter for the coming aws tutorials given we're building the venv up front), I think we'll want some requirements files for all these various options that enables building and running things from scratch.  also a good place limit the package versions to upstream changes are less likely to break things for users without us knowing

doutriaux1 committed:
added sina to pyranda study

edited plot_study and bayesian to get data from store

plot_study.py creates ensemble as well

doutriaux1 committed:
Merge branch 'main' into charles_edits

doutriaux1 committed:
need to have enum match actual query

doutriaux1 committed:
Update rayleigh_taylor_lc_build_venv_sim.yaml

doutriaux1 committed:
Update bayesian_inference.py

doutriaux1 committed:
addind exp file

bgunnar5 committed:
Merge pull request #4 from LLNL/charles_edits

Charles edits

bgunnar5 committed:
Merge branch 'pyranda_rt' into renee_updates
